shipyard-controller/db/migration/project_credentials_migration_test.go

```diff
@@ -0,0 +1,192 @@
+				GetOldCredentialsProjectsFunc: func() ([]*models.ExpandedProjectOld, error) {
+					return nil, fmt.Errorf("some err")
+				},
+			},
+			want: fmt.Errorf("could not transform git credentials to new format: some err"),
+		},
+		{
+			name: "update project err",
+			secretRepo: db_mock.MongoDBSecretCredentialsRepoMock{
+				UpdateSecretFunc: func(project *models.ExpandedProjectOld) error {
+					return nil
+				},
+			},
+			projectRepo: db_mock.MongoDBProjectCredentialsRepoMock{
+				GetOldCredentialsProjectsFunc: func() ([]*models.ExpandedProjectOld, error) {
+					return []*models.ExpandedProjectOld{
+						{
+							CreationDate:     "date",
+							LastEventContext: nil,
+							ProjectName:      "project",
+							Shipyard:         "shippy",
+							ShipyardVersion:  "ship",
+							Stages:           nil,
+							GitRemoteURI:     "http://some-url",
+							GitUser:          "user",
+							InsecureSkipTLS:  false,
+						},
+						{
+							CreationDate:     "date",
+							LastEventContext: nil,
+							ProjectName:      "project2",
+							Shipyard:         "shippy",
+							ShipyardVersion:  "ship",
+							Stages:           nil,
+							GitRemoteURI:     "http://some-url2",
+							GitUser:          "user",
+							InsecureSkipTLS:  false,
+						},
+					}, nil
+				},
+				UpdateProjectFunc: func(project *models.ExpandedProjectOld) error {
+					return fmt.Errorf("some err")
+				},
+			},
+			want: fmt.Errorf("could not transform git credentials for project project: some err"),
+		},
+		{
+			name: "update secret err",
+			secretRepo: db_mock.MongoDBSecretCredentialsRepoMock{
+				UpdateSecretFunc: func(project *models.ExpandedProjectOld) error {
+					return fmt.Errorf("some err")
+				},
+			},
+			projectRepo: db_mock.MongoDBProjectCredentialsRepoMock{
+				GetOldCredentialsProjectsFunc: func() ([]*models.ExpandedProjectOld, error) {
+					return []*models.ExpandedProjectOld{
+						{
+							CreationDate:     "date",
+							LastEventContext: nil,
+							ProjectName:      "project",
+							Shipyard:         "shippy",
+							ShipyardVersion:  "ship",
+							Stages:           nil,
+							GitRemoteURI:     "http://some-url",
+							GitUser:          "user",
+							InsecureSkipTLS:  false,
+						},
+						{
+							CreationDate:     "date",
+							LastEventContext: nil,
+							ProjectName:      "project2",
+							Shipyard:         "shippy",
+							ShipyardVersion:  "ship",
+							Stages:           nil,
+							GitRemoteURI:     "http://some-url2",
+							GitUser:          "user",
+							InsecureSkipTLS:  false,
+						},
+					}, nil
+				},
+			},
+			want: fmt.Errorf("could not transform git credentials for project project: some err"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			repo := ProjectCredentialsMigrator{
+				secretRepo:  tt.secretRepo,
+				projectRepo: tt.projectRepo,
+			}
+			err := repo.Transform()
+			require.Equal(t, tt.want, err)
```

Comment:
since migrating all projects always comes with some risk I would suggest to make the assertions more detailed to make sure no data is being lost or wrongly modified. Maybe you can use the sequence executions migrator as an orientation for that: https://github.com/keptn/keptn/blob/a9c1d2fc748198326af8981319d851a430c2c654/shipyard-controller/db/migration/sequence_execution_migration_test.go#L93 :)

Reply:
tests for transformation of the data are present in `project_credentials_repo_test.go` and `secret_credentials_repo_test.go` with hopefully all possible corner cases, there is no need to test it in this place again. This test aims to be more of a top level test, where the the used repositories are mocked